### PR TITLE
adding fuji75 pcb support

### PR DESCRIPTION
**Pull request by CMMS-Freather**
adding fuji75 pcb support
includes
solder
hotswap

## Description
adding fuji75 pcb support
includes
solder
hotswap

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/sakura_workshop/fuji75/solder/readme.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+Flashing example for this keyboard:
+
+    make sakura_workshop/fuji75/solder:default:flash
+
+## Bootloader Atmel-DFU
+
+For reset instruction, use the physical reset button on the back of the keyboard to enter bootloader mode
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```
Comment (zvecr) [inline keyboards/sakura_workshop/fuji75/solder/readme.md:21]:
```suggestion
See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).

## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB
* **Keycode in layout**: Press the key mapped to `QK_BOOT` if it is available
```

File: keyboards/sakura_workshop/fuji75/hotswap/readme.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+Flashing example for this keyboard:
+
+    make sakura_workshop/fuji75/hotswap:default:flash
+
+##Bootloader Atmel-DFU
+
+For reset instruction, use the physical reset button on the back of the keyboard to enter bootloader mode
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```
Comment (zvecr) [inline keyboards/sakura_workshop/fuji75/hotswap/readme.md:21]:
```suggestion
See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).

## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB
* **Keycode in layout**: Press the key mapped to `QK_BOOT` if it is available
```

File: keyboards/sakura_workshop/fuji75/solder/readme.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Fuji75 Solder
+
+![Fuji75 Solder](https://i.imgur.com/tv6YdWt.png)
+
```
Comment (zvecr) [inline keyboards/sakura_workshop/fuji75/solder/readme.md:4]:
```suggestion
```

File: keyboards/sakura_workshop/fuji75/hotswap/readme.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
+#Fuji Hotswap
+
+![Fuji hotswap](https://i.imgur.com/SfucJxa.png)
+
```
Comment (zvecr) [inline keyboards/sakura_workshop/fuji75/hotswap/readme.md:4]:
```suggestion
# Fuji75 Hotswap

![Fuji75 hotswap](https://i.imgur.com/SfucJxa.png)
```

File: keyboards/sakura_workshop/fuji75/config.h [hunk 5]
```diff
@@ -0,0 +1,8 @@
+// Copyright 2023 CMM.S Freather
```
Comment (zvecr) [inline keyboards/sakura_workshop/fuji75/config.h:1]: This file should be removed

File: keyboards/sakura_workshop/fuji75/info.json [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    "features": {
+        "bootmagic": true,
+        "mousekey": true,
+        "extrakey": true,
+        "nkro": true,
+        "rgblight": true
+    },
+    "rgblight": {
+        "led_count": 1,
+        "sleep": true,
```
Comment (zvecr) [inline keyboards/sakura_workshop/fuji75/info.json:17]:
```suggestion
       "layers": {
             "enabled": true
        },
        "sleep": true,
```

File: keyboards/sakura_workshop/fuji75/fuji75.c [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+void keyboard_post_init_user(void){
+  rgblight_layers = my_rgb_layers;
+}
+
+
+bool led_update_user(led_t led_state) {
+    rgblight_set_layer_state(0, led_state.caps_lock);
+    rgblight_set_layer_state(1, !led_state.caps_lock);
+    return true;
+}
```
Comment (zvecr) [inline keyboards/sakura_workshop/fuji75/fuji75.c:42]: `_user` functions at the keyboard level is invalid.

```suggestion
void keyboard_post_init_kb(void) {
  rgblight_layers = my_rgb_layers;
  
  keyboard_post_init_user();
}

bool led_update_kb(led_t led_state) {
    bool res = led_update_user(led_state);
    if(res) {
        rgblight_set_layer_state(0, led_state.caps_lock);
        rgblight_set_layer_state(1, !led_state.caps_lock);
    }
    return res;
}
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/sakura_workshop/fuji75/hotswap/readme.md [hunk 8]
```diff
@@ -0,0 +1,24 @@
+# Fuji75 Hotswap
+
+![Fuji75 hotswap](https://i.imgur.com/SfucJxa.png)
```
Comment (drashna) [inline keyboards/sakura_workshop/fuji75/hotswap/readme.md:3]:
```suggestion
![Fuji75 hotswap](https://i.imgur.com/SfucJxah.png)
```

File: keyboards/sakura_workshop/fuji75/solder/readme.md [hunk 9]
```diff
@@ -0,0 +1,24 @@
+# Fuji75 Solder
+
+![Fuji75 Solder](https://i.imgur.com/tv6YdWt.png)
```
Comment (drashna) [inline keyboards/sakura_workshop/fuji75/solder/readme.md:3]:
```suggestion
![Fuji75 Solder](https://i.imgur.com/tv6YdWth.png)
```

**Comment by CMMS-Freather** [comment]
@zvecr @drashna bump

**Review by drashna — APPROVED** [review]

**Comment by CMMS-Freather** [comment]
@drashna would you mind look again if it can be merged? it has been a while

File: keyboards/sakura_workshop/fuji75/hotswap/info.json
[the remark this answers is not in the document]
Comment (CMMS-Freather) [reply keyboards/sakura_workshop/fuji75/hotswap/info.json:1]: updated

File: keyboards/sakura_workshop/fuji75/solder/info.json
[the remark this answers is not in the document]
Comment (CMMS-Freather) [reply keyboards/sakura_workshop/fuji75/solder/info.json:1]: updated

**Review by fauxpark — APPROVED** [review]

**Review by waffle87 — APPROVED** [review]

**Pull request merged**